module/zfs/bqueue.c

```diff
@@ -87,13 +87,17 @@ bqueue_enqueue(bqueue_t *q, void *data, uint64_t item_size)
 void *
 bqueue_dequeue(bqueue_t *q)
 {
-	void *ret;
+	void *ret = NULL;
 	uint64_t item_size;
 	mutex_enter(&q->bq_lock);
 	while (q->bq_size == 0) {
 		cv_wait(&q->bq_pop_cv, &q->bq_lock);
 	}
 	ret = list_remove_head(&q->bq_list);
+	if (ret == NULL) {
```

Comment:
This function supposedly waits for a new element. Under what circumstances would list_remove_head() return NULL here? Should it go back to waiting instead?